spec/controllers/users_controller_spec.rb

```diff
@@ -315,6 +315,57 @@
     end
   end
 
+  describe '.admin_login' do
+    let(:admin) { Fabricate(:admin) }
+    let(:user) { Fabricate(:user) }
+
+    context 'enqueues mail' do
+      it 'enqueues mail with admin email and sso enabled' do
+        SiteSetting.enable_sso = true
+        Jobs.expects(:enqueue).with(:user_email, has_entries(type: :admin_login, user_id: admin.id))
+        put :admin_login, email: admin.email
+      end
+
+      it 'does not enqueue mail with admin email and sso disabled' do
+        SiteSetting.enable_sso = false
+        Jobs.expects(:enqueue).never
+        put :admin_login, email: admin.email
+      end
+
+      it 'does not enqueue mail with normal user email and sso enabled' do
+        SiteSetting.enable_sso = true
+        Jobs.expects(:enqueue).never
```

Comment:
in general I would just test for error code here returned by controller as opposed to magic job stuff.